tests/lib/source-code/token-store.js

```diff
@@ -654,6 +654,38 @@ describe("TokenStore", () => {
             assert.strictEqual(token.value, "c");
         });
 
+        it("should return null if the source contains only comments", () => {
+            const code = "// comment";
+            const ast = espree.parse(code, { loc: true, range: true, tokens: true, comment: true });
+            const tokenStore = new TokenStore(ast.tokens, ast.comments);
+            const token = tokenStore.getFirstToken(ast, {
+                filter() {
+                    assert.fail("Unexpected call to filter callback");
+                }
+            });
+
+            assert.strictEqual(token, null);
+        });
+
+        it("should return null if the source is empty", () => {
+            const code = "";
+            const ast = espree.parse(code, { loc: true, range: true, tokens: true, comment: true });
+            const tokenStore = new TokenStore(ast.tokens, ast.comments);
+            const token = tokenStore.getFirstToken(ast);
+
+            assert.strictEqual(token, null);
+        });
+
+        it("should retrieve a parent node's token if a node has no tokens", () => {
+            const code = "foo``";
+            const ast = espree.parse(code, { loc: true, range: true, tokens: true, comment: true, ecmaVersion: 6 });
+            const tokenStore = new TokenStore(ast.tokens, ast.comments);
+            const emptyTemplateElementNode = ast.body[0].expression.quasi.quasis[0];
+            const token = tokenStore.getFirstToken(emptyTemplateElementNode);
+
+            assert.strictEqual(token.value, "``");
+        });
```

Comment:
If a node has no tokens, I think `getFirstToken` should return `null`. Either way, this doesn't seem to be a case where a node has no tokens, because that token does belong to the TemplateElement node (`ast.body[0].expression.quasi.quasis[0]` node).

Reply:
Agree.

Reply:
@mdjermanovic you are correct! That node is not empty. I was fooled by [AST explorer](https://astexplorer.net/#/gist/02cd291223766f6e755f1208d5b137c2/117ef64f24b42ba58acb7c5cbc99b75a9cc43dd8) that shows the same value for both the `start` and `end` indices, probably because it uses an outdated version of Espree.
I added this test to make sure that I wouldn't inadvertently change the existing behavior, but now I think we can safely remove it because it's flawed.